qemu/tests/vioinput_keyboard.py

```diff
@@ -0,0 +1,136 @@
+import json
+import os
+import time
+import logging
+
+from virttest import error_context
+from virttest import utils_test
+from virttest import graphical_console
+from provider import input_event_proxy
+from virttest import data_dir
+
+
+@error_context.context_aware
+def keyboard_test(test, params, vm, key_name='', hold_time=0, interval=0):
+    """
+    Keyboard test. Support single key and combination key tests.
+
+    :param test: kvm test object
+    :param params: dictionary with the test parameters
+    :param vm: vm object
+    :param key_name: tested keyname, default key_name='', will covered all keys
+                     on the keyboard(us).
+    :param hold_time: the time that keep one key pressed, default hole_time=0.
+    :param interval: the interval between two keys, default interval=0.
+    """
+
+    def get_keycode_cfg(filename):
```

Comment:
please move it to the top level of this module.

Reply:
ok, thanks, will update~